### PR TITLE
Remove version related code from test generator

### PR DESCRIPTION
**Pull request by jpotts244**
This pull request removes all of the `version` and `bookkeeping` related code of the Exercism Ruby app. 

Changes include:
- Removal of `version` references in command line option.
- Removal of `version` generator unit tests.
- Removal of `version` and `bookkeeping` references in Generator source code.
- Removal of  `exercise/*/.meta/.version files`.
- Remove reference to `version` documentation.

This work is further described in the following Github issue: [Remove "version" related code from test generator](https://github.com/exercism/ruby/issues/840)

The work for this PR touched and deleted a lot of files, would love as much feedback as possible! There are a lot of changes included in here!

### CODE REVIEW

File: test/generator/command_line_test.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ def test_default_options
     def test_update_option
```
Comment (Insti) [inline test/generator/command_line_test.rb]: Removing this entire test is probably fine.

File: lib/generator/command_line.rb [hunk 2]
```diff
@@ -32,11 +32,7 @@ def generator(implementation)
     end
 
     def generator_class
-      update? ? UpdateVersionAndGenerateTests : GenerateTests
-    end
-
-    def update?
-      @options[:update]
+      GenerateTests
```
[the remark this answers is not in the document]
Comment (Insti) [reply lib/generator/command_line.rb]: Removing this wrapper method is a good idea.

File: test/generator/implementation_test.rb [hunk 3]
```diff
@@ -84,47 +52,4 @@ def test_add_2_numbers
       Object.send(:remove_const, :AlphaCase)
     end
   end
-
-  class LoggingImplementationTest < Minitest::Test
-    def test_build_tests
```
Comment (Insti) [inline test/generator/implementation_test.rb]: I think this test is still relevant?

**Comment by Insti** [comment]
Thanks @jpotts244 This all looks great :+1: 

Great pull request description. :heart:

I've had a quick look through it and left some comments. Let me know if you disagree with any of them, it's been a while since I last looked at the generator code.

I'm a bit busy for the next few days so probably won't get a chance to merge this until next week, but what you have done looks like it does all the right things. 

Good job. :star:

File: test/generator/implementation_test.rb (hunk 3 above)
Comment (jpotts244) [reply test/generator/implementation_test.rb]: That is very true! Fixing now 😅

**Comment by jpotts244** [comment]
Hi @Insti! Thanks so much for taking the time to review this PR! I've addressed you feedback - let me know what you think, no rush.

It has been super fun project, thanks for having me! Let me know if you'd like me to edit anything here or if I can help out in other ways!

**Comment by kotp** [comment]
I haven't extensively reviewed, but it does look good to me.  I hope this also serves as a bump for @Insti .  The best kind of PR, getting rid of lines of code! :)

File: test/generator_test.rb [hunk 4]
```diff
@@ -1,30 +0,0 @@
-require_relative 'test_helper'
-
-module Generator
-  class UpdateVersionAndGenerateTestsTest < Minitest::Test
```
Comment (Insti) [inline test/generator_test.rb]: This test is still required, or the `build_tests` call in `generator.rb` is never tested.
This test should be renamed though.

**Comment by Insti** [comment]
This method and its test can go:
```
lib/generator/files/track_files.rb:      def update_version(version)
test/generator/files/track_files_test.rb:      def test_update_version
```

**Comment by Insti** [comment]
There's also some options stuff that can go:
```
lib/generator/command_line/generator_optparser.rb:36:        parser.on('-u', '--update', 'Update test version') { |value| options[:update] = value }
lib/generator/command_line/generator_optparser.rb:37:        parser.on('-a', '--all', 'Regenerate all available test suites (does not update version)') do |value|
lib/generator.rb:14:  # Doesn't update the version information.
test/generator/command_line/generator_optparser_test.rb:28:    def test_update_option
test/generator/command_line/generator_optparser_test.rb:32:          default_options.merge(slug: 'beta', update: true),
```

File: test/generator_test.rb (hunk 4 above)
Comment (jpotts244) [reply test/generator_test.rb]: Taking a peek now!

**Comment by jpotts244** [comment]
Made some changes as you suggested!

File: lib/generator/command_line/generator_optparser.rb [hunk 5]
```diff
@@ -33,10 +33,6 @@ def parse_options
     def option_parser
       @option_parser ||= OptionParser.new do |parser|
         parser.banner = "Usage: #{$PROGRAM_NAME} [options] exercise-generator"
-        parser.on('-u', '--update', 'Update test version') { |value| options[:update] = value }
-        parser.on('-a', '--all', 'Regenerate all available test suites (does not update version)') do |value|
```
Comment (Insti) [inline lib/generator/command_line/generator_optparser.rb]: Oops, I guess I was not clear enough here: `--all` still needs to be an option, but the "does not update version" message can go.

Comment (Insti) [reply lib/generator/command_line/generator_optparser.rb]: I suggest reverting https://github.com/exercism/ruby/pull/877/commits/3e14ddf1611953108c569f44be2327dd077a4247 and trying again to just remove the `-u` options.

**Comment by Insti** [comment]
I found some more now-unused code and removed it because I wanted to try the Github code suggestion feature. I hope you don't mind.

File: lib/generator/command_line/generator_optparser.rb (hunk 5 above)
Comment (jpotts244) [reply lib/generator/command_line/generator_optparser.rb]: Hi there! sorry for my delay, sorry about that I'll update tonight!

**Comment by Insti** [comment]
Thanks a lot @jpotts244 that was a big one! :heart:

**Pull request merged**